data_steward/analytics/cdr_ops/ad_hoc_analyses/CT_RT_clean_ckecks.py

```diff
@@ -0,0 +1,277 @@
+group by person_id, measurement_source_concept_id, unit_concept_id,  measurement_concept_id, measurement_datetime, value_as_number,value_as_concept_id)A
+where A.rownum>1
+order by A.rownum desc
+'''
+execute(client, query)
+
+# # Records with 0 or null in specific *_concept_id fields should be dropped
+# Identifies if any records are not dropped based on the criteria in the doc string
+# of the drop_zero_concept_ids.py module.
+
+query = f'''
+with should_be_dropped as (
+    SELECT
+     COUNTIF((condition_source_concept_id is null or condition_source_concept_id = 0) AND
+             (condition_concept_id is null or condition_concept_id = 0)) as n_records_to_drop,
+     'condition_occurrence' as table_to_drop_from
+    FROM `{project_id}.{clean_dataset}.condition_occurrence`
+
+    UNION ALL
+
+    SELECT
+     COUNTIF((procedure_source_concept_id is null or procedure_source_concept_id = 0) AND
+             (procedure_concept_id is null or procedure_concept_id = 0)) as n_records_to_drop,
+     'procedure_occurrence' as table_to_drop_from
+    FROM `{project_id}.{clean_dataset}.procedure_occurrence`
+
+    UNION ALL
+
+    SELECT
+     COUNTIF((visit_source_concept_id is null or visit_source_concept_id = 0) AND
+             (visit_concept_id is null or visit_concept_id = 0)) as n_records_to_drop,
+     'visit_occurrence' as table_to_drop_from
+    FROM `{project_id}.{clean_dataset}.visit_occurrence`
+
+    UNION ALL
+
+    SELECT
+     COUNTIF((drug_source_concept_id is null or drug_source_concept_id = 0) AND
+             (drug_concept_id is null or drug_concept_id = 0)) as n_records_to_drop,
+     'drug_exposure' as table_to_drop_from
+    FROM `{project_id}.{clean_dataset}.drug_exposure`
+
+    UNION ALL
+
+    SELECT
+     COUNTIF((device_source_concept_id is null or device_source_concept_id = 0) AND
+             (device_concept_id is null or device_concept_id = 0)) as n_records_to_drop,
+     'device_exposure' as table_to_drop_from
+    FROM `{project_id}.{clean_dataset}.device_exposure`
+
+    UNION ALL
+
+    SELECT
+     COUNTIF((observation_source_concept_id is null or observation_source_concept_id = 0) AND
+             (observation_concept_id is null or observation_concept_id = 0)) as n_records_to_drop,
+     'observation' as table_to_drop_from
+    FROM `{project_id}.{clean_dataset}.observation`
+
+    UNION ALL
+
+    SELECT
+     COUNTIF((measurement_source_concept_id is null or measurement_source_concept_id = 0) AND
+             (measurement_concept_id is null or measurement_concept_id = 0)) as n_records_to_drop,
+     'measurement' as table_to_drop_from
+    FROM `{project_id}.{clean_dataset}.measurement`
+
+#    UNION ALL
+#
+#    SELECT
+#     COUNTIF((cause_source_concept_id is null or cause_source_concept_id = 0) AND
+#             (cause_concept_id is null or cause_concept_id = 0)) as n_records_to_drop,
+#     'death' as table_to_drop_from
+#    FROM `{project_id}.{clean_dataset}.death`
```

Comment:
@ksdkalluri go ahead and remove this code.  it is essentially dead.

Reply:
death is now dead.